src/main/java/com/innobridge/ethpay/service/TransactionService.java

```diff
@@ -94,5 +93,35 @@ public Transaction createTransaction(String senderId,
         return savedTransaction;
     }
 
+    public Transaction acceptTransaction(String transactionId) {
+        // TODO Auto-generated method stub
+        // Get transaction
+        Transaction transaction = transactionRepository.findById(transactionId)
+                .orElseThrow(() -> new IllegalArgumentException("Transaction not found: " + transactionId));
+
+        accountService.updatePendingTransaction(transaction.senderId, transaction.sourceCurrency);
```

Comment:
update pending transaction should be done after transactionRepository.save(transaction);